pyproject.toml

```diff
@@ -1,3 +1,43 @@
+[build-system]
+requires = ["poetry-core"]
+build-backend = "poetry.core.masonry.api"
+
+[tool.poetry]
+name = "qibotn"
+version = "0.0.1"
+description = ""
+authors = ["Qibo Team"]
+readme = "README.md"
+
+[tool.poetry.dependencies]
+python = "^3.9,<3.13"
+quimb = { version = "^1.6.0", extras = ["tensor"] }
```

Comment:
Thanks for implementing the poetry, Alessandro. 

I tried a bit locally and found missing dependencies to conduct tests locally. Can you add `qibo` and `qibojit` dependencies? They are the minimal additional dependencies needed for the quimb test to pass. 

Also, please change the highest Python version to <3.12. Otherwise, you will not be able to install Qibo. 

I noticed Cupy and cuQuantum have been parked under cuda dependencies. Thus, I would assume they will be treated as separate installations as described in the installation note for Qibojit.

Reply:
> I tried a bit locally and found missing dependencies to conduct tests locally. Can you add `qibo` and `qibojit` dependencies? They are the minimal additional dependencies needed for the quimb test to pass.

At some point, it was said that `qibotn` should not depend on `qibojit`, because it's kind of parallel.

> Also, please change the highest Python version to <3.12. Otherwise, you will not be able to install Qibo.

But for sure I'm still missing the dep on Qibo :)

> I noticed Cupy and cuQuantum have been parked under cuda dependencies. Thus, I would assume they will be treated as separate installations as described in the installation note for Qibojit.

Kind of: at the moment it's in the wrong position, as a development group (a Poetry feature, only present when managing the env through Poetry) and not as a `pip` extra (also possible with Poetry, and persisting through packaging).
I will change the status (I had a problem installing cuda in my local env, that was the main reason for the temporary issue).